omniqueue/src/backends/azure_queue_storage.rs

```diff
@@ -0,0 +1,200 @@
+        let client = get_client(&config);
+        Ok(AQSQueueProducer { client, config })
+    }
+
+    async fn consuming_half(config: Self::Config) -> Result<AQSQueueConsumer> {
+        let client = get_client(&config);
+        Ok(AQSQueueConsumer { client, config })
+    }
+}
+
+pub struct AQSQueueProducer {
+    client: QueueClient,
+    config: AQSQueueConfig,
+}
+
+impl AQSQueueProducer {
+    pub async fn send_raw(&self, payload: &str) -> Result<()> {
+        self.send_raw_scheduled(payload, Duration::ZERO).await
+    }
+
+    pub async fn send_raw_scheduled(&self, payload: &str, delay: Duration) -> Result<()> {
+        self.client
+            .put_message(payload)
+            .visibility_timeout(delay)
+            .ttl(self.config.message_ttl)
+            .into_future()
+            .await
+            .map_err(QueueError::generic)
+            .map(|_| ())
+    }
+
+    pub async fn send_serde_json<P: Serialize + Sync>(&self, payload: &P) -> Result<()> {
+        let payload = serde_json::to_string(payload)?;
+        self.send_raw(&payload).await
+    }
+    pub async fn send_serde_json_scheduled<P: Serialize + Sync>(
+        &self,
+        payload: &P,
+        delay: Duration,
+    ) -> Result<()> {
+        let payload = serde_json::to_string(payload)?;
+        self.send_raw_scheduled(&payload, delay).await
+    }
+}
+
+impl_queue_producer!(AQSQueueProducer, String);
+impl_scheduled_queue_producer!(AQSQueueProducer, String);
+
+pub struct AQSQueueConsumer {
+    client: QueueClient,
+    config: AQSQueueConfig,
+}
+
+struct AQSAcker {
+    client: QueueClient,
+    already_acked_or_nacked: bool,
+    pop_receipt: PopReceipt,
+}
+
+#[async_trait]
+impl Acker for AQSAcker {
+    async fn ack(&mut self) -> Result<()> {
+        if self.already_acked_or_nacked {
+            return Err(QueueError::CannotAckOrNackTwice);
+        }
+        self.already_acked_or_nacked = true;
+        self.client
+            .pop_receipt_client(self.pop_receipt.clone())
+            .delete()
+            .into_future()
+            .await
+            .map_err(QueueError::generic)
+            .map(|_| ())
+    }
+
+    async fn nack(&mut self) -> Result<()> {
+        Ok(())
+    }
+}
+
+impl AQSQueueConsumer {
+    fn wrap_message(&self, message: &Message) -> Delivery {
+        Delivery {
+            acker: Box::new(AQSAcker {
+                client: self.client.clone(),
+                pop_receipt: message.pop_receipt(),
+                already_acked_or_nacked: false,
+            }),
+            payload: Some(message.message_text.as_bytes().to_owned()),
+        }
+    }
+
+    /// Note that blocking receives are not supported by Azure Queue Storage
+    /// and will return immediately if no messages are available for delivery
+    /// in the queue.
+    pub async fn receive(&mut self) -> Result<Delivery> {
+        self.client
+            .get_messages()
+            .into_future()
+            .await
```

Comment:
If this `.into_future` is not an inherent method, it's redundant. `.await` automatically calls `IntoFuture::into_future`.